hircluster.c

```diff
@@ -2302,8 +2274,19 @@ static void *redis_cluster_command_execute(redisClusterContext *cc,
         goto error;
     }
 
+    /* If update slotmap has been scheduled, do that in the same pipeline. */
+    if (cc->need_update_route && c_updating_route == NULL) {
+        if (clusterUpdateRouteSendCommand(cc, c) == REDIS_OK) {
+            c_updating_route = c;
+        }
+    }
+
     if (redisGetReply(c, &reply) != REDIS_OK) {
         __redisClusterSetError(cc, c->err, c->errstr);
+        /* We may need to update the slotmap if this node is removed from the
+         * cluster, but the current request may have already timed out so we
+         * schedule it for later. */
+        cc->need_update_route = 1;
```

Comment:
```suggestion
        if (c->err != REDIS_ERR_OOM)
          cc->need_update_route = 1;
```
Maybe we shouldn't trigger more work if hiredis gives OOM here?
We could just return the OOM-failure to the user so it can be handled accordingly, and then let next non-OOM failing command trigger a slotmap update. Not sure whats best in OOM cases..

Additionally this fixes the OOM testcase which sets the cluster in a bad state (but which the client should handle I guess..) and most testcase changes can then be reverted. It's only the "no reachable node in cluster" checks that needs to be changed then.